os/binfmt/binfmt_execmodule.c

```diff
@@ -333,12 +272,14 @@ int exec_module(FAR struct binary_s *binp)
 
 	pid = tcb->cmn.pid;
 
-#ifdef CONFIG_APP_BINARY_SEPARATION
 	rtcb->uheap = 0;
+#ifdef CONFIG_ARM_MPU						/* MPU register values for loadable apps only */
+	memset(rtcb->mpu_regs, 0, sizeof(rtcb->mpu_regs));
```

Comment:
Could you leave why those two lines are necessary?

Reply:
Done